File: WorkOutGenerator/app/src/main/java/com/sport/workoutgenerator/RetrofitClient.java
package com.sport.workoutgenerator;

import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;

public class RetrofitClient {
    private static final String BASE_URL = "http://159.69.90.204/api/WorkOutGenerator/";
    private static Retrofit retrofit;

    public static ApiService createService(Class<ApiService> apiServiceClass) {
        if (retrofit == null) {
            retrofit = new Retrofit.Builder()
                    .baseUrl(BASE_URL)
                    .addConverterFactory(GsonConverterFactory.create())
                    .build();
        }
        return retrofit.create(ApiService.class);
    }
}

File: KtSportApp/app/src/main/java/com/sport/ktsportapp/RetrofitClient.kt
package com.sport.ktsportapp

import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory

object RetrofitClient {

    private const val BASE_URL = "http://159.69.90.204/api/KtSportApp/"

    val apiService: ApiService by lazy {
        Retrofit.Builder()
            .baseUrl(BASE_URL)
            .addConverterFactory(GsonConverterFactory.create())
            .build()
            .create(ApiService::class.java)
    }
}
